coffeecard/CoffeeCard.WebApi/Pages/Recover.cshtml.cs

```diff
@@ -39,12 +47,15 @@ public async Task<IActionResult> OnPostAsync()
         {
             if (!ModelState.IsValid) return Page();
 
-            if (await _accountService.RecoverUserAsync(Token, PinCode.NewPinCode))
+            try
             {
-                TempData["resultHeader"] = "Success";
-                TempData["result"] = "Your password has now been reset";
+                if (await _accountService.RecoverUserAsync(Token, PinCode.NewPinCode))
+                {
+                    TempData["resultHeader"] = "Success";
+                    TempData["result"] = "Your password has now been reset";
+                }
             }
-            else
+            catch (Exception e)

```

Comment:
Why the try/catch here, what is the scenario for this?

Reply:
Same reason as above. The is catch for any unhandled Exceptions from the service. Otherwise, if an exception does occur, it will in the best case be propagated to the user as a plain json website, and worst case bring down the entire site (Happened once in development).